### PR TITLE
gather_nd Op refine

### PR DESCRIPTION
**Pull request by wangchaochaohu**
### PR types

Function optimization
### PR changes

APIs
### Describe

refine the paddle.gather_nd

### CODE REVIEW

**Review by heavengate — APPROVED** [review]

**Review by ForFishes — APPROVED** [review]
lgtm

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -8266,14 +8266,18 @@ def gather_nd(input, index, name=None):
                          = [23]
 
     Args:
-        input (Variable): The source input. Its dtype should be int32, int64, float32, float64.
-        index (Variable): The index input with rank > 1, index.shape[-1] <= input.rank.
-                          Its dtype should be int32, int64.
-        name (str|None): A name for this layer(optional). If set None, the
-                         layer will be named automatically.
+        input (Tensor): The source input. Its dtype should be bool, float32, float64, int32, int64.
```
Comment (XiaoguangHu01) [inline python/paddle/fluid/layers/nn.py:8269]: fluid.layers.gather_nd 补充一个deprecated标记，记录API之间的对应关系。

**Review by XiaoguangHu01 — APPROVED** [review]
LTGM

File: python/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -1135,3 +1135,88 @@ def reshape(x, shape, name=None):
...
+
+        output[(i_0, ..., i_{K-2})] = input[index[(i_0, ..., i_{K-2})]]
+
+    Obviously, :code:`index.shape[-1] <= input.rank` . And, the output tensor has
+    shape :code:`index.shape[:-1] + input.shape[index.shape[-1]:]` .
+
+    .. code-block:: text
+
+            Given:
+                input = [[[ 0,  1,  2,  3],
```
Comment (XiaoguangHu01) [inline python/paddle/tensor/manipulation.py:1160]: input -> x

**Review by TCChenlong — APPROVED** [review]
LGTM

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (wangchaochaohu) [reply python/paddle/fluid/layers/nn.py:8269]: https://github.com/PaddlePaddle/Paddle/pull/26579 已加

File: python/paddle/tensor/manipulation.py (hunk 2 above)
Comment (wangchaochaohu) [reply python/paddle/tensor/manipulation.py:1160]: https://github.com/PaddlePaddle/Paddle/pull/26579 done

**Pull request merged**